tool/generator/lib/src/projection/utils.dart

```diff
@@ -341,6 +345,217 @@ String parseGenericTypeIdentifierName(TypeIdentifier typeIdentifier) {
+      return 'u4';
+    case BaseType.uint64Type:
+      return 'u8';
+    default:
+      throw UnsupportedError('Unsupported baseType: $baseType');
+  }
+}
+
+/// Returns the type signature for the given [td].
+String parseTypeDefSignature(TypeDef td) {
+  if (td.typeSpec != null) return parseTypeIdentifierSignature(td.typeSpec!);
+
+  if (td.isClass) {
+    final className = td.name;
+    final defaultInterfaceSignature =
+        parseTypeDefSignature(td.interfaces.first);
+    return 'rc($className;$defaultInterfaceSignature)';
+  }
+
+  return td.guid!.toLowerCase();
+}
+
+/// Returns the type signature for the given [ti].
+String parseTypeIdentifierSignature(TypeIdentifier ti) {
+  if (ti.baseType == BaseType.genericTypeModifier) {
+    return parseGenericTypeIdentifierSignature(ti);
+  }
+
+  if (ti.name == 'System.Guid') return 'g16';
+
+  final typeProjection = TypeProjection(ti);
+
+  if (typeProjection.isWinRTDelegate) {
+    return 'delegate(${ti.type!.guid!.toLowerCase()})';
+  }
+
+  if (typeProjection.isWinRTEnum) {
+    final enumName = ti.name;
+    final isFlagsEnum = ti.type!.existsAttribute('System.FlagsAttribute');
+    final enumSignature = isFlagsEnum ? 'u4' : 'i4';
+    return 'enum($enumName;$enumSignature)';
+  }
+
+  if (typeProjection.isWinRTStruct) {
+    final structName = ti.name;
+    final fields = ti.type!.fields
+        .map((field) => parseTypeIdentifierSignature(field.typeIdentifier));
+    return 'struct($structName;${fields.join(';')})';
+  }
+
+  if (typeProjection.isInterface) return ti.type!.guid!.toLowerCase();
+
+  if (typeProjection.isClass) {
+    final className = ti.name;
+    final defaultInterface = ti.type!.interfaces.first;
+    if (defaultInterface.typeSpec != null) {
+      final defaultInterfaceSignature =
+          parseTypeIdentifierSignature(defaultInterface.typeSpec!);
+      return 'rc($className;$defaultInterfaceSignature)';
+    }
+
+    final defaultInterfaceSignature = parseTypeDefSignature(defaultInterface);
+    return 'rc($className;$defaultInterfaceSignature)';
+  }
+
+  return primitiveTypeSignatureFromBaseType(ti.baseType);
+}
+
+/// Returns the type signature for the given generic [typeIdentifier].
+String parseGenericTypeIdentifierSignature(TypeIdentifier typeIdentifier) {
+  if (typeIdentifier.baseType != BaseType.genericTypeModifier) {
+    throw ArgumentError('Expected a generic type identifier.');
+  }
+
+  final parentTypeGuid = typeIdentifier.type!.guid!.toLowerCase();
+
+  if (typeIdentifier.type?.genericParams.length == 2) {
+    final firstArgSignature =
+        parseTypeIdentifierSignature(typeIdentifier.typeArg!);
+    final secondArgSignature =
+        parseTypeIdentifierSignature(typeIdentifier.typeArg!.typeArg!);
+    return 'pinterface($parentTypeGuid;$firstArgSignature;$secondArgSignature)';
+  }
+
+  return 'pinterface($parentTypeGuid;${parseTypeIdentifierSignature(typeIdentifier.typeArg!)})';
+}
+
+const wrtPinterfaceNamespace = [
```

Comment:
What do these numbers represent?

Reply:
They are bytes that represent Guid `11f47ad5-7b73-42c0-abae-878b1e16adee`. 

![wrt_pinterface_namespace](https://user-images.githubusercontent.com/45524029/201889063-cea03502-e12a-4e05-a654-e521a7b0cfb5.png)
It is defined in https://learn.microsoft.com/en-us/uwp/winrt-cref/winrt-type-system#guid-generation-for-parameterized-types